cmake/modules/FindGEOS.cmake

```diff
@@ -0,0 +1,181 @@
+# Find GEOS
+# ~~~~~~~~~
+# Copyright (c) 2008, Mateusz Loskot <mateusz@loskot.net>
+# (based on FindGDAL.cmake by Magnus Homann)
+# Redistribution and use is allowed according to the terms of the BSD license.
+# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+#
+# CMake module to search for GEOS library
+#
+# If it's found it sets GEOS_FOUND to TRUE
+# and following variables are set:
+#    GEOS_INCLUDE_DIR
+#    GEOS_LIBRARY
+#
+
+#INCLUDE (${CMAKE_SOURCE_DIR}/cmake/MacPlistMacros.cmake)
+
+IF(WIN32)
+
+  IF (MINGW)
+    FIND_PATH(GEOS_INCLUDE_DIR geos_c.h /usr/local/include /usr/include c:/msys/local/include)
+    FIND_LIBRARY(GEOS_LIBRARY NAMES geos_c PATHS /usr/local/lib /usr/lib c:/msys/local/lib)
+  ENDIF (MINGW)
+
+  IF (MSVC)
+    FIND_PATH(GEOS_INCLUDE_DIR geos_c.h HINTS ${GEOS_DIR} ${GEOS_INCLUDE_DIR} $ENV{LIB_DIR}/include $ENV{INCLUDE})
+    FIND_LIBRARY(GEOS_LIBRARY NAMES geos geos_c_i geos_c  HINTS ${GEOS_DIR} PATHS
+      "$ENV{LIB_DIR}/lib"
+      $ENV{LIB}
+      )
```

Comment:
As I understand it, find_library will try to find `geos` first on Windows. Perhaps this is why the CI on Win finds goes instead of geos_c?

Reply:
Yes, looks very likely, though the same logic should apply to unix.

Reply:
Not the same. I think on unix it runs `geos-config` to get the library path.

Reply:
OK, so please try this order ``geos_c_i geos_c geos``. The *_i* is something that some use to mean *import_lib* which is what those libs are.